### PR TITLE
Add library for sending requests over knc

### PR DESCRIPTION
**Pull request by sdukhovni**

### CODE REVIEW

File: building/build-debs/homeworld-keyserver/src/keycommon/knc/knc.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package knc
+
+import (
+	"fmt"
+	"os/exec"
+)
+
+func Request(host string, data []byte) ([]byte, error) {
```
Comment (celskeggs) [inline building/build-debs/homeworld-keyserver/src/keycommon/knc/knc.go]: Can this be a reqtarget implementation? Or at least be wrapped in one?

File: building/build-debs/homeworld-keyserver/src/keycommon/knc/knc.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	go func() {
+		defer stdin.Close()
+		stdin.Write(data)
+	}()
+
+	response, err := cmd.Output()
```
Comment (celskeggs) [inline building/build-debs/homeworld-keyserver/src/keycommon/knc/knc.go]: How do timeouts and other errors show up in this output? Are they converted to errors?

Comment (sdukhovni) [reply building/build-debs/homeworld-keyserver/src/keycommon/knc/knc.go]: According to the golang docs, if the command exits unsuccessfully, Cmd.Output() will return an ExitError containing a snippet of stderr.

Comment (celskeggs) [reply building/build-debs/homeworld-keyserver/src/keycommon/knc/knc.go]: Can you confirm that this is the case?

Comment (sdukhovni) [reply building/build-debs/homeworld-keyserver/src/keycommon/knc/knc.go]: Yup, confirmed, the returned error is non-nil and a snippet of stderr is stored in `err.(*exec.ExitError).Stderr` as a bytestring.

File: building/build-debs/homeworld-keyserver/src/keycommon/knc/knc.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	raw_resps, err := k.kncRequest(raw_reqs)
+	if err != nil {
+		return nil, err
+	}
+
+	resps := []string{}
+	err = json.Unmarshal(raw_resps, &resps)
+	if err != nil {
+		return nil, err
+	}
```
Comment (celskeggs) [inline building/build-debs/homeworld-keyserver/src/keycommon/knc/knc.go]: You should verify that the correct number of results come back -- i.e., that len(reqs) == len(resps).

File: building/build-debs/homeworld-keyserver/src/keycommon/knc/knc.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import (
+	"fmt"
+	"os/exec"
+	"encoding/json"
+	"keycommon/reqtarget"
+)
+
+type KncServer struct {
+	Hostname string
+}
```
Comment (celskeggs) [inline building/build-debs/homeworld-keyserver/src/keycommon/knc/knc.go]: Can you provide a method on Keyserver (AuthenticateWithKerberosTickets) to construct an instance of this?

File: building/build-debs/homeworld-keyserver/src/keycommon/knc/knc.go (hunk 3 above)
Comment (sdukhovni) [reply building/build-debs/homeworld-keyserver/src/keycommon/knc/knc.go]: Fixed

File: building/build-debs/homeworld-keyserver/src/keycommon/knc/knc.go (hunk 4 above)
Comment (sdukhovni) [reply building/build-debs/homeworld-keyserver/src/keycommon/knc/knc.go]: Done

**Pull request merged**